internal/util/file/file_test.go

```diff
@@ -98,3 +98,51 @@ func TestCreateTempFile_WithLargeContent(t *testing.T) {
 		t.Fatalf("Content mismatch: got %v, want %v", string(readContent), content)
 	}
 }
+
+func TestCreateSparseFile(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name    string
+		sizeMB  int64
+		wantErr bool
+	}{
+		{"WithValidSize", 10, false},
+		{"WithZeroSize", 0, true},
+		{"WithNegativeSize", -1, true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			file, err := os.CreateTemp("", "test-sparse-")
```

Comment:
place this in the `t.TempDir()` directory for automatic cleanup, no need to have to remove the file in a `defer`.

Reply:
Rebased and fixed this nit for you :-)